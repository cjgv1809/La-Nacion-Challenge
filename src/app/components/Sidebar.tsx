function Sidebar() {
  return (
    <aside className="sidebar__aside">
      <div className="banner --desktop --large"></div>
      <div className="com-ranking hlp-none hlp-tablet-none">
        <h2 className="com-title-section-m">Recetas más leídas</h2>
      </div>
      <div className="banner --desktop --large"></div>
    </aside>
  );
}

export default Sidebar;
