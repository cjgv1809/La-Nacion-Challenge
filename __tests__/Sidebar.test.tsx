import { render, screen } from "@testing-library/react";
import Sidebar from "@/app/components/layout/Sidebar";

describe("Sidebar", () => {
  it("renders the sidebar", () => {
    render(<Sidebar />);

    const sidebar = screen.getByRole("complementary");

    expect(sidebar).toBeInTheDocument();
  });

  it("renders a heading for the most read recipes", () => {
    render(<Sidebar />);

    const heading = screen.getByRole("heading", { level: 2 });

    expect(heading).toHaveTextContent("Recetas más leídas");
  });

  it("renders a heading with a different text", () => { 
    render(<Sidebar />);

    const heading = screen.getByRole("heading", { level: 2 });

    expect(heading).not.toHaveTextContent("Recetas más populares");
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<Sidebar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
