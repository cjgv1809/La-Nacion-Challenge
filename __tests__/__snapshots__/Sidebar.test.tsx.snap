// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar matches the snapshot 1`] = `
<DocumentFragment>
  <aside
    class="sidebar__aside"
  >
    <div
      class="banner --desktop --large"
    />
    <div
      class="com-ranking hlp-none hlp-tablet-none"
    >
      <h2
        class="com-title-section-m"
      >
        Recetas más leídas
      </h2>
    </div>
    <div
      class="banner --desktop --large"
    />
  </aside>
</DocumentFragment>
`;
